### ISSUE TITLE
Incremental Learning?

### ISSUE THREAD

**Issue by jmrichardson** [timeseriesAI/tsai#169]
Hi, I am new to this project and looks great!  I have a question regarding minirocket.  Following along the tutorial, I believe I understand how to use tsai.  However, I have a ton of data that are in multiple files that I would like to process each step by step and not combine into a large numpy array (too much memory).  I saw your article on memmap but prefer not to combine them into a large numpy array even on disk.  Rather, is there a way to incrementally learn?  Meaning, I process each file into a numpy array (in tsai format) and then hand over tsai, then iterate again on next file.  At the end, have a trained model?  I know that rocket requires normalization so not sure if it's even possible?  Thanks for any advice

**Comment by oguiza** [comment]
Hi @jmrichardson,
That's a great question.
Something to bear in mind is that MiniRocket is not a deep learning model. You use 10_000 convolutions, but they have requires_grad=False. That is they don't learn anything. Learning only happens in the head. And the head of the model is just a linear model defined by the number of classes. 
So based on that, you could use the same MiniRocket model, and replace the head. But you would need to train it on a sample set that is representative of your test set.

**Comment by jmrichardson** [comment]
Thank you and looking forward to using tsai!
